src/docfx/build/metadata/MetadataProvider.cs

```diff
@@ -17,7 +17,7 @@ internal class MetadataProvider
         private readonly List<(Func<string, bool> glob, string key, JToken value)> _rules = new List<(Func<string, bool>, string, JToken)>();
         private readonly List<(Func<string, bool> glob, string key, JToken value)> _monikerRules = new List<(Func<string, bool>, string, JToken)>();
 
-        private readonly ConcurrentDictionary<FilePath, UserMetadata> _metadataCache = new ConcurrentDictionary<FilePath, UserMetadata>();
+        private readonly ConcurrentDictionary<FilePath, Watch<UserMetadata>> _metadataCache = new ConcurrentDictionary<FilePath, Watch<UserMetadata>>();
```

Comment:
>_metadataCache [](start = 77, length = 14)

If there is no changes to the opConfig, since the docset is extracted from the Opconfig, the docsetBuilder won't be re-created, so this cache will will be reused, event if the file is changed, the cached metadata will be returned?

Reply:
The value of this cache will change :)